components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOFacade.java

```diff
@@ -0,0 +1,313 @@
+    }
+
+    private void encryptAuthenticationSecrets(ActionDTO actionDTO) throws ActionMgtException {
+
+        try {
+            List<AuthProperty> encryptedProperties = actionSecretProcessor.encryptAssociatedSecrets(
+                    actionDTO.getEndpoint().getAuthentication(), actionDTO.getId());
+            actionDTO.setAuthenticationProperties(encryptedProperties);
+        } catch (SecretManagementException e) {
+            throw new ActionMgtServerException("Error while encrypting Action Endpoint Authentication Secrets.", e);
+        }
+    }
+
+    private void updateAuthenticationSecrets(ActionDTO updatingActionDTO, ActionDTO existingActionDTO)
+            throws ActionMgtException {
+
+        if (updatingActionDTO.getEndpoint() == null || updatingActionDTO.getEndpoint().getAuthentication() == null) {
+            return;
+        }
+
+        Authentication updatingAuthentication = updatingActionDTO.getEndpoint().getAuthentication();
+        Authentication existingAuthentication = existingActionDTO.getEndpoint().getAuthentication();
+
+        try {
+            if (updatingAuthentication.getType() != existingAuthentication.getType()) {
+                actionSecretProcessor.deleteAssociatedSecrets(existingAuthentication, existingActionDTO.getId());
+            }
+            List<AuthProperty> encryptedProperties = actionSecretProcessor.encryptAssociatedSecrets(
+                    updatingAuthentication, updatingActionDTO.getId());
+            updatingActionDTO.setAuthenticationProperties(encryptedProperties);
+        } catch (SecretManagementException e) {
+            throw new ActionMgtServerException("Error while updating Action Endpoint Authentication Secrets.", e);
+        }
+    }
+
+    private void deleteAuthenticationSecrets(ActionDTO deletingActionDTO) throws ActionMgtServerException {
+
+        try {
+            actionSecretProcessor.deleteAssociatedSecrets(deletingActionDTO.getEndpoint().getAuthentication(),
+                    deletingActionDTO.getId());
+        } catch (SecretManagementException e) {
+            throw new ActionMgtServerException("Error while deleting Action Endpoint Authentication Secrets.", e);
+        }
+    }
+
+    private void addProperties(ActionDTO actionDTO, Integer tenantId) throws ActionPropertyResolverException {
+
+        Map<String, String> properties = null;
+        ActionPropertyResolver actionPropertyResolver =
+                ActionPropertyResolverFactory.getActionPropertyResolver(actionDTO.getType());
+
+        if (actionPropertyResolver != null) {
+            properties = actionPropertyResolver.addProperties(actionDTO,
+                    IdentityTenantUtil.getTenantDomain(tenantId));
+        }
+        if (properties != null) {
+            actionDTO.setProperties(properties.entrySet().stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+        }
```

Comment:
IMO, the objective of the resolver should be to populate the properties.
So either let them to return properties or let them to populate in the transferred DTO. Should choose one path only

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6119/commits/edeb975693380b404a45f24842338951d17c1695